### PR TITLE
Fix pydocstyle D208 (Docstring is over-indented)

### PR DESCRIPTION
**Pull request by timhoffm**
## PR Summary

Fix all occurences of [pydocstyle D208: Docstring is over-indented](http://www.pydocstyle.org/en/stable/error_codes.html).

### CODE REVIEW

File: lib/matplotlib/colors.py [hunk 1]
```diff
@@ -613,16 +613,20 @@ def set_bad(self, color='k', alpha=None):
             self._set_extremes()
 
     def set_under(self, color='k', alpha=None):
-        """Set color to be used for low out-of-range values.
-           Requires norm.clip = False
+        """
+        Set color to be used for low out-of-range values.
```
Comment (anntzer) [inline lib/matplotlib/colors.py]: "Set the color for low out-of-range values when ``norm.clip = False``." would fit in one line.

File: lib/matplotlib/testing/jpl_units/UnitDblConverter.py [hunk 2]
```diff
@@ -28,8 +28,8 @@ def rad_fn(x, pos=None):
 
 
 class UnitDblConverter(units.ConversionInterface):
-    """: A matplotlib converter class.  Provides matplotlib conversion
-          functionality for the Monte UnitDbl class.
+    """
+    Provides matplotlib conversion functionality for the Monte UnitDbl class.
```
Comment (anntzer) [inline lib/matplotlib/testing/jpl_units/UnitDblConverter.py]: capitalize matplotlib
"Monte"???

File: lib/matplotlib/testing/jpl_units/UnitDblFormatter.py [hunk 3]
```diff
@@ -6,24 +6,26 @@
...
-    """The formatter for UnitDbl data types.  This allows for formatting
-        with the unit string.
+    """
+    The formatter for UnitDbl data types.
+
+    This allows for formatting with the unit string.
     """
     def __init__(self, *args, **kwargs):
-        'The arguments are identical to matplotlib.ticker.ScalarFormatter.'
+        """The arguments are identical to matplotlib.ticker.ScalarFormatter."""
```
Comment (anntzer) [inline lib/matplotlib/testing/jpl_units/UnitDblFormatter.py]: this docstring and the following ones could just be inherited (in fact the whole `__init__` can be deleted and inherited...).

Comment (timhoffm) [reply lib/matplotlib/testing/jpl_units/UnitDblFormatter.py]: Thanks! Removed all.

File: lib/matplotlib/dates.py [hunk 4]
```diff
@@ -325,8 +325,14 @@ def __init__(self, fmt):
...
-        """s : string to be converted
-           return value: a date2num float
+        """
+        Parameters
+        ----------
+        s : str
+
+        Returns
+        -------
+        num : date2num float
```
Comment (anntzer) [inline lib/matplotlib/dates.py]: I would not invent a name for the return value here (per https://numpydoc.readthedocs.io/en/latest/format.html you can just specify the type without a name).

File: lib/matplotlib/__init__.py [hunk 5]
```diff
@@ -1251,21 +1251,24 @@ def interactive(b):
     """
     Set interactive mode to boolean b.
```
Comment (anntzer) [inline lib/matplotlib/__init__.py]: Something like "Set whether to draw after every plotting command (such as `~.Axes.xlabel`)." fits in a single line (replacing the entire docstring).
`

File: lib/matplotlib/__init__.py [hunk 6]
```diff
@@ -1251,21 +1251,24 @@ def interactive(b):
...
 
-    If b is True, then draw after every plotting command, e.g., after xlabel
+    If b is True, then draw after every plotting command, e.g., after xlabel.
     """
     rcParams['interactive'] = b
 
 
 def is_interactive():
-    'Return true if plot mode is interactive'
```
Comment (anntzer) [inline lib/matplotlib/__init__.py]: "... whether to redraw after each command", as for `interactive()` (the current docstring is pretty uninformative)

File: lib/matplotlib/colors.py [hunk 7]
```diff
@@ -613,16 +613,16 @@ def set_bad(self, color='k', alpha=None):
             self._set_extremes()
 
     def set_under(self, color='k', alpha=None):
-        """Set color to be used for low out-of-range values.
-           Requires norm.clip = False
+        """
+        Set the color low out-of-range values when ``norm.clip = False``.
```
Comment (anntzer) [inline lib/matplotlib/colors.py]: *for* low

File: lib/matplotlib/testing/jpl_units/EpochConverter.py [hunk 8]
```diff
@@ -10,8 +10,9 @@
 
 
 class EpochConverter(units.ConversionInterface):
-    """: A matplotlib converter class.  Provides matplotlib conversion
-          functionality for Monte Epoch and Duration classes.
+    """
+    Provides matplotlib conversion functionality for Monte Epoch and Duration
```
Comment (anntzer) [inline lib/matplotlib/testing/jpl_units/EpochConverter.py]: Capitalize here too.

**Comment by anntzer** [comment]
anyone can merge postci

**Pull request merged**